gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/BaseDistributionPlugin.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.palantir.gradle.dist.tasks.CreateManifestTask;
+import groovy.lang.Closure;
+import java.io.File;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Optional;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.util.GFileUtils;
+
+public final class BaseDistributionPlugin implements Plugin<Project> {
```

Comment:
Base is possibly a bit misleading here, as this literally only adds a single `getMinimumProductVersion` utility... can we try some alternate names?